packages/front-end/pages/metrics.tsx

```diff
@@ -371,122 +371,130 @@ const MetricsPage = (): React.ReactElement => {
-              onClick={(e) => {
-                e.preventDefault();
-                router.push(getMetricLink(metric.id));
-              }}
-              style={{ cursor: "pointer" }}
-              className={metric.archived ? "text-muted" : ""}
-            >
-              <td>
-                <Link href={getMetricLink(metric.id)}>
-                  <a
-                    className={`${
-                      metric.archived ? "text-muted" : "text-dark"
-                    } font-weight-bold`}
-                  >
-                    <MetricName id={metric.id} />
-                  </a>
-                </Link>
-              </td>
-              <td>{metric.type}</td>
-
-              <td className="col-4">
-                <SortedTags
-                  tags={metric.tags ? Object.values(metric.tags) : []}
-                  shouldShowEllipsis={true}
-                />
-              </td>
-              <td className="col-2">
-                {metric && (metric.projects || []).length > 0 ? (
-                  <ProjectBadges
-                    resourceType="metric"
-                    projectIds={metric.projects}
-                    className="badge-ellipsis short align-middle"
-                  />
-                ) : (
-                  <ProjectBadges
-                    resourceType="metric"
-                    className="badge-ellipsis short align-middle"
-                  />
-                )}
-              </td>
-              <td>{metric.owner}</td>
-              <td className="d-none d-lg-table-cell">
-                {metric.datasourceName}
-                {metric.datasourceDescription && (
-                  <div
-                    className="text-gray font-weight-normal small text-ellipsis"
-                    style={{ maxWidth: 350 }}
-                  >
-                    {metric.datasourceDescription}
-                  </div>
-                )}
-              </td>
-              <td
-                title={datetime(metric.dateUpdated || "")}
-                className="d-none d-md-table-cell"
-              >
-                {metric.managedBy === "config"
-                  ? ""
-                  : ago(metric.dateUpdated || "")}
-              </td>
-              <td className="text-muted">
-                {metric.archived && (
-                  <Tooltip
-                    body={"Archived"}
-                    innerClassName="p-2"
-                    tipMinWidth="auto"
-                  >
-                    <FaArchive />
-                  </Tooltip>
-                )}
-              </td>
-              <td
-                style={{ cursor: "initial" }}
+          {items.map((metric) => {
+            const showMoreMenu: boolean =
+              (metric.onDuplicate && editMetricsPermissions[metric.id]) ||
+              (!metric.managedBy &&
+                metric.onArchive &&
+                editMetricsPermissions[metric.id]) ||
+              false;
```

Comment:
Previously, we were always showing the `<MoreMenu/>` component, but in a lot of cases, the modal that opened as a result was empty. This resolves that, so we only show it if there is at least 1 item inside.

Reply:
This is going to be hard to maintain.  A better approach is to create an array of MoreMenu links.  If that array is empty, then skip rendering the menu.

```tsx
const moreMenuLinks: ReactElement[] = [];

if (metric.onDuplicate && editMetricsPermissions[metric.id]) {
  moreMenuLinks.push(<button key="duplicate" ...>)
}
```

Reply:
oof - yes, much better pattern. Updating now.